tests/unit/h/services/annotation_sync_test.py

```diff
@@ -0,0 +1,336 @@
+        queue_service.get.return_value = []
+
+        def add_job(indexed=True, updated=False, deleted=False, **kwargs):
+            annotation = factories.Annotation()
+            job = factories.SyncAnnotationJob(annotation=annotation, **kwargs)
+            queue_service.get.return_value.append(job)
+
+            if indexed:
+                index(annotation)
+
+            if updated:
+                annotation.updated = now + datetime.timedelta(weeks=1)
+
+            if deleted:
+                annotation.deleted = True
+
+        add_job()
+        add_job(indexed=False)
+        add_job(updated=True)
+        add_job(deleted=True)
+        add_job(tag="tag_2", force=True)
+
+        counts = svc.sync(5)
+
+        assert counts == {
+            "Synced/Total": 3,
+            "Completed/Total": 3,
+            "Synced/test_tag/Total": 2,
+            "Completed/test_tag/Total": 2,
+            "Synced/test_tag/Different_in_Elastic": 1,
+            "Synced/test_tag/Missing_from_Elastic": 1,
+            "Synced/tag_2/Forced": 1,
+            "Synced/tag_2/Total": 1,
+            "Completed/tag_2/Forced": 1,
+            "Completed/tag_2/Total": 1,
+            "Completed/test_tag/Up_to_date_in_Elastic": 1,
+            "Completed/test_tag/Deleted_from_db": 1,
+        }
+
+    def url_safe_id(self, job):
+        """Return the URL-safe version of the given job's annotation ID."""
+        return URLSafeUUID.hex_to_url_safe(job.kwargs["annotation_id"])
+
+    @pytest.fixture
+    def index(
+        self,
+        pyramid_request,
+        es_client,
+        moderation_service,  # pylint:disable=unused-argument
+        nipsa_service,  # pylint:disable=unused-argument
+    ):
+        """Return a method that indexes an annotation into Elasticsearch."""
+
+        # Construct a real (not mock) SearchIndexService so we can call its
+        # methods to index annotations.
+        #
+        # This isn't ideal because it means these AnnotationSyncService
+        # unit tests are coupled to SearchIndexService and any other services
+        # or code that SearchIndexService calls.
+        #
+        # On the other hand, this avoids these tests having to duplicate
+        # SearchIndexService's code for indexing annotations and ensures that
+        # the documents in the search index are the same as they would be in
+        # production.
+        #
+        # (FIXME: The real solution to this issue is to refactor the services
+        # to be more coherent and avoid this testing dilemma.)
+        search_index_service = SearchIndexService(
+            request=pyramid_request,
+            es=es_client,
+            settings={},
+            annotation_read_service=sentinel.annotation_read_service,
+        )
+
+        def index(annotation):
+            """Index `annotation` into Elasticsearch."""
+            search_index_service.add_annotation(annotation)
+            es_client.conn.indices.refresh(index=es_client.index)
```

Comment:
The previous tests were already doing this but `SearchIndexService.add_annotation()` was part of the class under test since the `sync()` method was in `SearchIndexService` as well. There will be a better way of decoupling this but maybe we can leave it for another day, since I have limited time